qemu/tests/virtio_fs_share_data.py

```diff
@@ -0,0 +1,97 @@
+    cmd_mount_target = params.get('cmd_mount_target')
+
+    cmd_pjdfstest = params.get('cmd_pjdfstest')
+    pjdfstest_pkg = params.get('pjdfstest_pkg')
+
+    fio_options = params.get('fio_options')
+    io_timeout = params.get_numeric('io_timeout')
+
+    username = params.get('username')
+    password = params.get('password')
+    port = params.get('file_transfer_port')
+
+    vm = env.get_vm(params.get("main_vm"))
+    vm.verify_alive()
+    session = vm.wait_for_login()
+    host_addr = vm.get_address()
+
+    for fs in params.objects("filesystems"):
+        fs_params = params.object_params(fs)
+        fs_target = fs_params.get("fs_target")
+        fs_dest = fs_params.get("fs_dest")
+
+        fs_source = fs_params.get("fs_source_dir")
+        base_dir = fs_params.get("fs_source_base_dir", data_dir.get_data_dir())
+        if not os.path.isabs(fs_source):
+            fs_source = os.path.join(base_dir, fs_source)
+        guest_data = os.path.join(fs_dest, 'fs_test')
+        host_data = os.path.join(fs_source, 'fs_test')
+
+        error_context.context("Create a destination directory %s "
+                              "inside guest." % fs_dest, logging.info)
+        session.cmd(cmd_create_dest_dir % fs_dest)
+
+        error_context.context("Mount virtiofs target %s to %s inside guest."
+                              % (fs_target, fs_dest), logging.info)
+        session.cmd(cmd_mount_target % (fs_target, fs_dest))
+
+        if cmd_dd:
+            logging.info("Creating file under %s inside guest." % fs_dest)
+            session.cmd(cmd_dd % guest_data, io_timeout)
+            logging.info("Compare the md5 between guest and host.")
+            md5_guest = session.cmd(cmd_md5 % guest_data,
+                                    io_timeout).strip().split()[0]
+            logging.info(md5_guest)
+            md5_host = process.run(cmd_md5 % host_data,
+                                   io_timeout).stdout_text.strip().split()[0]
+            if md5_guest != md5_host:
+                test.fail('The md5 value of host is not same to guest.')
+
+        if fio_options:
+            error_context.context("Run fio on %s." % fs_dest, logging.info)
+            fio = generate_instance(params, vm, 'fio')
+            try:
+                fio.run(fio_options % guest_data, io_timeout)
+            finally:
+                fio.clean()
+            vm.verify_dmesg()
+
+        if cmd_pjdfstest:
+            error_context.context("Run pjdfstest on %s." % fs_dest, logging.info)
+            host_path = os.path.join(data_dir.get_deps_dir('pjdfstest'), pjdfstest_pkg)
+            scp_to_remote(host_addr, port, username, password, host_path, fs_dest)
+            session.cmd(params.get('cmd_unpack') % fs_dest, 180)
+            session.cmd(params.get('cmd_yum_deps'), 180)
+            session.cmd(params.get('cmd_make') % fs_dest, io_timeout)
+            session.cmd(cmd_pjdfstest % fs_dest, io_timeout)
```

Comment:
check the result and fail the test accordingly ?

Reply:
Yes, I need it